python/pyarrow/includes/libarrow.pxd

```diff
@@ -1199,6 +1199,60 @@ cdef extern from "arrow/api.h" namespace "arrow" nogil:
     shared_ptr[CScalar] MakeNullScalar(shared_ptr[CDataType] type)
 
 
+cdef extern from "arrow/dlpack_structure.h" nogil:
+    cdef enum DLDeviceType:
```

Comment:
I think this can be removed now (or at least part of it) now the bulk of the implementation doesn't live in cython?

Reply:
Just need to push, have it locally 👍

Reply:
Sorry, misunderstood this one 😊

Reply:
I think the structs and `DLDeviceType` all have to stay as they are interconnected and so I only removed `DLDataTypeCode`: https://github.com/apache/arrow/pull/38472/commits/3c35fb37ceeecbbd615daeae030060e1b2e20edf

Reply:
Done: https://github.com/apache/arrow/pull/38472/commits/ee6c7fac3e6a150bf8a3cad4a6c1f15121144194